posthog/models/event_property.py

```diff
@@ -0,0 +1,49 @@
+from django.contrib.postgres.indexes import GinIndex
+from django.db import models
+from django.utils import timezone
+
+from posthog.models.team import Team
+
+
+class EventProperty(models.Model):
+    class PropertyType(models.TextChoices):
+        NUMBER = "NUMBER", "Number"
+        STRING = "STRING", "String"
+        BOOLEAN = "BOOLEAN", "Boolean"
+        DATETIME = "DATETIME", "DateTime"
+
+    team: models.ForeignKey = models.ForeignKey(
+        # "event_properties" is a field on Team, so have to call this "event_properties_model"
+        Team,
+        on_delete=models.CASCADE,
+        related_name="event_properties_model",
+        related_query_name="team",
+    )
+    event: models.CharField = models.CharField(max_length=400, null=False)
+    property: models.CharField = models.CharField(max_length=400, null=False)
+
+    # contains e.g. the date format
+    property_type: models.CharField = models.CharField(
+        max_length=20, choices=PropertyType.choices, default=PropertyType.STRING, null=True
+    )
+    property_type_format: models.CharField = models.CharField(max_length=100, null=True)
```

Comment:
This should probably also be an enum, but waiting for feedback on the different types to put here... or if to even keep.

Reply:
I think we should definitely store, not least so that we can give people the ability to manually set a type

Since we're putting it on the property and not even definition I'd be tempted to not add it here until we actually need it though. Otherwise we need a pit of success for future travellers to fall into so they know where to store what